pandas/tests/test_frame.py

```diff
@@ -7559,6 +7559,20 @@ def test_info_duplicate_columns(self):
                           columns=['a', 'a', 'b', 'b'])
         frame.info(buf=io)
 
+    def test_info_duplicate_columns_shows_correct_dtypes(self):
+        io = StringIO()
+
+        # it works!
+        frame = DataFrame([[1, 2.0]],
+                          columns=['a', 'a'])
+        frame.info(buf=io)
+        expected_dtype1 = 'a    1 non-null int64\n'
```

Comment:
you can do this as a single line comparison
just ``self.assertEqual(io.getvalue(),'a    1 .....')